packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -61,19 +70,53 @@ export class ConnectionStateHandler extends EventEmitterWithErrorHandling<IConne
         super();
     }
 
-    public receivedAddMemberEvent(clientId: string, quorum: Quorum) {
+    // This is true when this client submitted any ops.
+    public setDirtyState() {
+        this.isDirty = true;
+    }
+
+    public receivedAddMemberEvent(clientId: string) {
         // This is the only one that requires the pending client ID
         if (clientId === this.pendingClientId) {
+            // Wait for previous client to leave the quorum before firing "connected" event.
+            if (this.prevClientLeftTimer !== undefined && this.prevClientLeftTimer.hasTimer) {
```

Comment:
I do not see a reason to differentiate this.prevClientLeftTimer === undefined & !this.prevClientLeftTimer.hasTimer.
If so, please create timer in constructor and change type to never be undefined.
Alternatively you could change that to this.prevClientLeftTimer?.hasTimer here and in other places, but I'd prefer less states

Reply:
Added in constructor.